src/main/java/com/mraof/minestuck/item/TempleScannerItem.java

```diff
@@ -0,0 +1,72 @@
+package com.mraof.minestuck.item;
+
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.*;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraft.world.World;
+import net.minecraft.world.server.ServerWorld;
+
+public class TempleScannerItem extends Item
+{
+	public TempleScannerItem(Properties properties)
+	{
+		super(properties);
+	}
+	
+	@Override
+	public ActionResult<ItemStack> onItemRightClick(World worldIn, PlayerEntity playerIn, Hand handIn)
+	{
+		worldIn.playSound(playerIn, playerIn.getPosX(), playerIn.getPosY(), playerIn.getPosZ(), SoundEvents.UI_BUTTON_CLICK, SoundCategory.AMBIENT, 0.8F, 1.3F);
+		
+		ItemStack item = playerIn.getHeldItem(handIn);
+		ItemStack uraniumStack = new ItemStack(MSItems.RAW_URANIUM);
+		boolean foundItem = false;
+		
+		for(ItemStack invItem : playerIn.inventory.mainInventory)
+		{
+			if(ItemStack.areItemsEqual(invItem, uraniumStack))
+			{
+				foundItem = true;
+				if(random.nextFloat() >= 0.95F)
```

Comment:
the playSound function inside that section works clientside, if you can recommend a workaround I will adopt it!

Reply:
When playSound is called on both sides, the player should be provided so that for that player, the sound is only played from client-side
When playSound is called only on server-side, null should be provided as the player so that the sound is played from server-side for all players

Reply:
Resolved!